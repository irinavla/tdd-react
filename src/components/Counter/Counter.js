import React, { Component } from 'react'

class Counter extends Component {

  constructor(props) {
    super(props);

    this.state = {
      count: 0
    }
  }

  increment = () => {
    this.setState( prevState => {
      return {
        count: prevState.count + 1
      }
    })
  }

  decrement = () => {
    this.setState( prevState => {
      return {
        count: prevState.count - 1
      }
    })
  }

  render() {
    return (
        <div className="row">
          <div className="col s12 m6 offset-m3">
            <div className="card grey lighten-3">
              <div className="card-content">
                <span className="card-title">Counter component</span>
                {/* TODO: get count from state here */}
                <p className="counter-index">Count is: </p>
              </div>
              <div className="card-action">
                {/* TODO: setup increment/decrement methods here */}
                <button className="btn waves-effect waves-light pink darken-3 add">Add</button>
                <button className="btn waves-effect waves-light pink darken-3 subtract">Remove</button>
              </div>
            </div>
          </div>
        </div>
    )
  }
}

export default Counter