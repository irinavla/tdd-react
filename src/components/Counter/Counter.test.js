import React from 'react'
import Counter from './Counter'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Counter', () => {
  // we use the it() function to run individual tests

  // test the initial state
  test('starts with a count of 0', () => {
    // render the component and get an instance of it
    const container = shallow(<Counter />);
    const count = container.find('p.counter-index').text();
    expect(count).toEqual('Count is: 0');
  });

  // test user interactions, i.e. click events
  test('increment count when button is clicked', () => {
    const container = shallow(<Counter />);
    const incrementBtn = container.find('button.add');
    incrementBtn.simulate('click');

    const count = container.find('p.counter-index').text();
    expect(count).toEqual('Count is: 1');
  });

  test('decrement count when button is clicked', () => {
    const container = shallow(<Counter />);
    const decrementBtn = container.find('button.subtract');
    decrementBtn.simulate('click');

    const count = container.find('p.counter-index').text();
    expect(count).toEqual('Count is: -1');
  });
}) 


/* 
https://jestjs.io/docs/en/api#testname-fn-timeout
https://jestjs.io/docs/en/expect#expectvalue
*/