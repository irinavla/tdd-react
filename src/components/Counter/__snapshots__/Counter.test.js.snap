// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component matches the snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m6 offset-m3"
  >
    <div
      className="card grey lighten-3"
    >
      <div
        className="card-content"
      >
        <span
          className="card-title"
        >
          Counter component
        </span>
        <p
          className="counter-index"
        >
          Count is: 
          0
        </p>
      </div>
      <div
        className="card-action"
      >
        <button
          className="btn waves-effect waves-light pink darken-3 add"
          onClick={[Function]}
        >
          Add
        </button>
        <button
          className="btn waves-effect waves-light pink darken-3 subtract"
          onClick={[Function]}
        >
          Remove
        </button>
      </div>
    </div>
  </div>
</div>
`;
